packages/amplify-cli-core/API.md

```diff
@@ -183,6 +222,54 @@ export enum AmplifyFrontend {
     javascript = "javascript"
 }
 
+// @public (undocumented)
+export class AmplifyInternalOnlyPostEnvRemoveEventData extends AmplifyEventData {
+    constructor(envName: string);
+    // (undocumented)
+    readonly envName: string;
+}
+
+// @public (undocumented)
+export class AmplifyPostCodegenModelsEventData extends AmplifyEventData {
+}
+
+// @public (undocumented)
+export class AmplifyPostEnvAddEventData extends AmplifyEventData {
+    constructor(prevEnvName: string, newEnvName: string);
+    // (undocumented)
+    readonly newEnvName: string;
+    // (undocumented)
+    readonly prevEnvName: string;
+}
+
+// @public (undocumented)
+export class AmplifyPostInitEventData extends AmplifyEventData {
+}
+
+// @public (undocumented)
+export class AmplifyPostPullEventData extends AmplifyEventData {
+}
+
+// @public (undocumented)
+export class AmplifyPostPushEventData extends AmplifyEventData {
+}
+
+// @public (undocumented)
+export class AmplifyPreCodegenModelsEventData extends AmplifyEventData {
+}
+
+// @public (undocumented)
+export class AmplifyPreInitEventData extends AmplifyEventData {
+}
+
+// @public (undocumented)
+export class AmplifyPrePullEventData extends AmplifyEventData {
+}
+
+// @public (undocumented)
+export class AmplifyPrePushEventData extends AmplifyEventData {
+}
```

Comment:
I wonder why do we have `AmplifyEvent` enum and all these types. 
Should we fix this while doing this refactor?

Reply:
@sobolk I pushed an update that, in my opinion, simplifies the interface. The no-argument events all share a common interface now, and the relationship between the events and the data they required is now codified in the type system rather than just by naming convention. @edwardfoyle would like your eyes on this as well if you have a chance.